Core/Src/main.c

```diff
@@ -87,14 +91,45 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
+  // {ADC start, CH0, don't care}
+  uint8_t tx_data[COMMS_LENGTH] = {0b0000001, 0b1000000, 0b0000000};
+  uint8_t rx_data[COMMS_LENGTH] = {};
+
+  // Start the timer once
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
+
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+
+    // Set CS to low
+    HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);
+    // Transmit and receive data
+    HAL_SPI_TransmitReceive(&hspi1, tx_data, rx_data, sizeof(tx_data), 100);
+    // Set CS back to high
+    HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
+
+    // Gives a 16 bit value (MSB) where only 10 bits are relevant (0x3FF = 0b1111111111)
+    uint16_t adc_rx_value = (rx_data[1] << 8 | rx_data[1]) && 0x3FF;
+
+    // Use adc_rx_value to calculate duty cycle
+    // D_min + (ADC/ADC_MAX) * (D_max - D_min)
+    float duty_cycle = (float) (0.05 * (1 + adc_rx_value/ADC_MAX_VALUE));
```

Comment:
Since `adc_rx_value` is declared as `uint16_t`, the division here `adc_rx_value/ADC_MAX_VALUE` will result in 0. Need to convert the type to floating point value before proceeding with the calculation.

Reply:
Will it result in 0 because I declared duty_cycle as float and the types don't match?